src/core/viz/parser.js

```diff
@@ -179,3 +179,78 @@ function cleanJsep() {
     jsep.addLiteral('true');
     jsep.addLiteral('false');
 }
+
+/**
+ * Remove comments from string
+ * - // line comments
+ * - /* block comments
+ * - ' " Keep comments inside single and double quotes
+ * Based on: https://j11y.io/javascript/removing-comments-in-javascript/
+ */
+export function cleanComments(str) {
+    var mode = {
+        singleQuote: false,
+        doubleQuote: false,
+        blockComment: false,
+        lineComment: false
+    };
+
+    // Adding chars to avoid index checking
+    str = ('_' + str + '_').split('');
```

Comment:
What's the reason for splitting and working with an array of one-char strings`str=['1', '2', '3']` instead of a string `str='123'` ?. In both cases we can index individual characters as `str[i]`, right?

Reply:
Now I see some characters get eliminated in the loop with `str[i]=''`

Reply:
It is because of string [i] is read-only.